server/src/main/java/org/opensearch/action/support/clustermanager/TransportClusterManagerNodeAction.java

```diff
@@ -134,12 +134,10 @@ protected boolean localExecute(Request request) {
 
     @Override
     protected void doExecute(Task task, final Request request, ActionListener<Response> listener) {
-        ClusterState state = clusterService.state();
-        logger.trace("starting processing request [{}] with cluster state version [{}]", request, state.version());
         if (task != null) {
             request.setParentTask(clusterService.localNode().getId(), task.getId());
         }
-        new AsyncSingleAction(task, request, listener).doStart(state);
+        new AsyncSingleAction(task, request, listener).start();
```

Comment:
what are we doing for cases like below: 

https://github.com/opensearch-project/OpenSearch/blob/3a97d4cad65fa19126340aabf7e8bd2b1ce2b822/server/src/main/java/org/opensearch/cluster/action/shard/ShardStateAction.java#L192

Here on the sender side , TransportClusterManagerNodeAction is not extended .

Reply:
Good point, I tried to scan over the code and found similar for `NodeMappingRefreshAction` as well. Let me know if I am missing from somewhere else as well.

We will need to plug the retry logic over here as well. Once we have finalized approach on it using `RetryableAction`/`ThrottlingRetryListener` will add that over here as well. 

Till then will keep this comment open for tracking.